src/oop/special_class_method/hard/q6/MainTest.java

```diff
@@ -0,0 +1,345 @@
+        return Stream.of(
+                Arguments.of(4535.4, "Long"),
+                Arguments.of(234.4, "Round"),
+                Arguments.of(654.45, "Second"),
+                Arguments.of(43.3, "Fourth")
+
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("constructorInputProvider")
+    @Tag("dependent1")
+    public void carClassHasCorrectConstructor(double value1, String value2) throws Throwable {
+        Object[][] arguments = {
+                {value1, double.class},
+                {value2, String.class}
+        };
+        Class<?>[] constructorClasses = {double.class, String.class};
+        Object classInstance = testClass.createInstance(arguments);
+        String incorrectDoubleValueMessage =
+                "Your " + className + " constructor with input does not initialize the " + doubleFieldName + " attribute to the correct value.";
+        String incorrectStringValueMessage =
+                "Your " + className + " constructor with input does not initialize the " + stringFieldName + " attribute to the correct value.";
+        String incorrectConstructorVisibilityModifier =
+                "Your " + className + " constructor does not have the correct visibility modifier.";
+        _assertEquals(value1, testClass.getFieldValue(classInstance, doubleFieldName),
+                incorrectDoubleValueMessage);
+        _assertEquals(value2, testClass.getFieldValue(classInstance, stringFieldName),
+                incorrectStringValueMessage);
+        assertTrue(testClass.hasModifier(constructorClasses, "public"), incorrectConstructorVisibilityModifier);
+    }
+
+    @Test
+    @Tag("dependent2")
+    public void carClassHasCorrectConstructor2() throws Throwable {
+        Object classInstance = testClass.createInstance();
+        String incorrectDoubleValueMessage =
+                "Your " + className + " constructor without input does not initialize the " + doubleFieldName + " attribute to the correct value.";
+        String incorrectStringValueMessage =
+                "Your " + className + " constructor without input does not initialize the " + stringFieldName + " attribute to the correct value.";
+        _assertEquals(180.0, testClass.getFieldValue(classInstance, doubleFieldName),
+                incorrectDoubleValueMessage);
+        assertEquals(null, testClass.getFieldValue(classInstance, stringFieldName),
+                incorrectStringValueMessage);
+    }
+
+    private static Stream<Arguments> doubleInputProvider() {
+        return Stream.of(
+                Arguments.of(4535.4, 432.2),
+                Arguments.of(234.4, 655.3),
+                Arguments.of(654.45, 453.3),
+                Arguments.of(43.3, 654.2)
+
+        );
+    }
+
+    @Test
+    public void setTopSpeedIsDefinedCorrectly() {
+        String incorrectMethodDefinition = "The %s method in the %s class is not defined correctly. Make sure it is declared, spelt correctly, and has the correct parameters.";
```

Comment:
this all already gets checked in the test below, I don't believe there is a need for this test. Same with the rest of the following "isDefinedCorrectly" tests